tests/dualtor/test_standby_tor_downstream.py

```diff
@@ -0,0 +1,193 @@
+        logger.info("Bring back interface {}".format(link_to_up))
+        dut.no_shutdown(link_to_up)
+
+
+@pytest.fixture
+def shutdown_one_uplink(rand_selected_dut):
+    link_to_shutdown = shutdown_random_one_t1_link(rand_selected_dut)
+    yield
+    no_shutdown_t1_link(rand_selected_dut, link_to_shutdown)
+
+
+def shutdown_random_one_bgp_session(dut):
+    """
+    Shutdown a random BGP session
+    """
+    bgp_facts = dut.get_bgp_neighbors()
+    up_bgp_neighbors = []
+    for k, v in bgp_facts.items():
+        if v['state'] == 'established' and ipaddress.ip_address(k).version == 4:
+            up_bgp_neighbors.append(k)
+    if not up_bgp_neighbors:
+        return None
+    bgp_to_shutdown = random.choice(up_bgp_neighbors)
+    logger.info("Shutting down bgp session with {}".format(bgp_to_shutdown))
+    dut.shell("config bgp shutdown neighbor {}".format(bgp_to_shutdown))
+    return bgp_to_shutdown
+
+
+def startup_bgp_session(dut, bgp_to_up):
+    """
+    Startup a BGP session
+    """
+    if bgp_to_up:
+        logger.info("Bring back bgp session with {}".format(bgp_to_up))
+        dut.shell("config bgp startup neighbor {}".format(bgp_to_up))
+
+
+@pytest.fixture
+def shutdown_one_bgp_session(rand_selected_dut):
+    bgp_shutdown = shutdown_random_one_bgp_session(rand_selected_dut)
+    yield
+    startup_bgp_session(rand_selected_dut, bgp_shutdown)
+
+
+def add_loopback_routes(standby_tor, active_tor_loopback_ip):
+    """
+    Add static routes to reach the peer's loopback.
+    The function is similar with fixture apply_dual_tor_peer_switch_route, but we can't use the fixture directly
+    """
+    logger.info("Applying dual ToR peer switch loopback route")
+    bgp_neighbors = standby_tor.bgp_facts()['ansible_facts']['bgp_neighbors'].keys()
+    
+    ipv4_neighbors = []
+
+    for neighbor in bgp_neighbors:
+        if ipaddress.ip_address(neighbor).version == 4:
+            ipv4_neighbors.append(neighbor)
+
+    nexthop_str = ''
+    for neighbor in ipv4_neighbors:
+        nexthop_str += 'nexthop via {} '.format(neighbor)
+
+    # Use `ip route replace` in case a rule already exists for this IP
+    # If there are no pre-existing routes, equivalent to `ip route add`
+    standby_tor.shell('ip route replace {}/32 {}'.format(active_tor_loopback_ip, nexthop_str))
+
+
+def remove_loopback_routes(standby_tor, active_tor_loopback_ip):
+    """
+    Remove static routes for active tor's loopback
+    """
+    logger.info("Removing dual ToR peer switch loopback route")
+    standby_tor.shell('ip route del {}/32'.format(active_tor_loopback_ip), module_ignore_errors=True)
+
+
+def test_standby_tor_downstream(dualtor_info):
+    """
+    Verify tunnel traffic to active ToR is distributed equally across nexthops, and
+    no traffic is forwarded to server from standby ToR
+    """
+    params = dualtor_info
+    check_tunnel_balance(**params)
+
+
+def test_standby_tor_downstream_one_t1_link_down(request, shutdown_one_uplink):
```

Comment:
This case seems redundant, since in the `test_standby_tor_downstream_t1_link_recovered` test case we also shutdown a link to a T1, it may be better if we change the `test_standby_tor_downstream_t1_link_recovered` case to include this one?

Reply:
Fixed. Thanks